spring-data-mongodb/src/test/java/org/springframework/data/mongodb/core/UncustomizableFindPublisher.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.mongodb.core;
+
+import com.mongodb.CursorType;
+import com.mongodb.reactivestreams.client.FindPublisher;
+import lombok.RequiredArgsConstructor;
+import org.bson.conversions.Bson;
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * {@link FindPublisher} implementation which does nothing in all FindPublisher's eigen methods, but delegates
+ * {@link #subscribe(Subscriber)} call to a {@link Publisher} passed in the constructor.
+ *
+ * @author Roman Puchkovskiy
+ */
+@RequiredArgsConstructor
+class UncustomizableFindPublisher<T> implements FindPublisher<T> {
```

Comment:
Doesn't it rather make sense to use a proper Mockito mock instead of introducing a `FindPublisher` implementation? This approach requires us to catch up each time MongoDB driver introduces any changes to `FindPublisher`.

Reply:
To be honest, with the number of methods there are on this interface, I thought it would be a nightmare to mock them all. But it looks like all the needed methods are already mocked. I've applied your advice and it seems working ok still being succinct.